web-ui/jsoo/main.ml

```diff
@@ -0,0 +1,80 @@
+module Dom_html = Js_of_ocaml.Dom_html
+module Dom = Js_of_ocaml.Dom
+module Js = Js_of_ocaml.Js
+module Firebug = Js_of_ocaml.Firebug
+
+(* module WebSockets = Js_of_ocaml.WebSockets *)
+module WebSocket = Brr_io.Websocket
+module Ev = Brr.Ev
+
+let regexp_left_paren = Re.Str.regexp_string "("
+let regexp_right_paren = Re.Str.regexp_string ")"
+let encoded_left_paren = "%28"
+let encoded_right_paren = "%29"
+
+let encode_parens s : Jstr.t =
+  let s' = Jstr.to_string s in
+  let s_lp_encoded =
+    Re.Str.global_replace regexp_left_paren encoded_left_paren s'
+  in
+  let s_rp_encoded =
+    Re.Str.global_replace regexp_right_paren encoded_right_paren s_lp_encoded
+  in
+  Jstr.of_string s_rp_encoded
+
+(* let js_str = Js.string *)
+let document = Dom_html.document
+
+let inject_log_lines data =
+  let new_div = Dom_html.createDiv document in
```

Comment:
At least on Safari, adding this extra `div` around the logs creates ugly spacing between the log lines. I'm unsure why.

<img width="338" alt="Screenshot 2023-03-22 at 11 49 44" src="https://user-images.githubusercontent.com/13054139/226882288-0e9125bd-79d6-47b1-80db-f14fad3270f8.png">